src/transport_ofi.h

```diff
@@ -658,6 +658,139 @@ void shmem_transport_put_nb(shmem_transport_ctx_t* ctx, void *target, const void
     }
 }
 
+static inline
+void shmem_transport_put_signal_nbi(shmem_transport_ctx_t* ctx, void *target, const void *source, size_t len,
+                                    uint64_t *sig_addr, uint64_t signal, int pe)
+{
+    int ret = 0;
+    uint64_t dst = (uint64_t) pe;
+    uint64_t polled = 0;
+    uint64_t key;
+    uint8_t *addr;
+
+    shmem_transport_ofi_get_mr(target, pe, &addr, &key);
+
+    if (len <= shmem_transport_ofi_max_buffered_send) {
+        uint8_t *src_buf = (uint8_t *) source;
+
+        SHMEM_TRANSPORT_OFI_CTX_LOCK(ctx);
+        SHMEM_TRANSPORT_OFI_CNTR_INC(&ctx->pending_put_cntr);
+
+        const struct iovec msg_iov = {
+                                       .iov_base = src_buf,
+                                       .iov_len = len
+                                     };
+        const struct fi_rma_iov rma_iov = {
+                                            .addr = (uint64_t) addr,
+                                            .len = len,
+                                            .key = key
+                                          };
+        const struct fi_msg_rma msg = {
+                                        .msg_iov = &msg_iov,
+                                        .desc = NULL,
+                                        .iov_count = 1,
+                                        .addr = GET_DEST(dst),
+                                        .rma_iov = &rma_iov,
+                                        .rma_iov_count = 1,
+                                        .context = src_buf,
+                                        .data = 0
+                                      };
+
+        do {
+            ret = fi_writemsg(ctx->ep, &msg, FI_DELIVERY_COMPLETE);
```

Comment:
Do you mean to add the `FI_INJECT` flag here (this is the buffered send case)?

Reply:
Yes, `FI_INJECT` should be added here. Thanks.